model/host/host.go

```diff
@@ -1330,10 +1342,10 @@ func (h *Host) Terminate(ctx context.Context, user, reason string) error {
 	return nil
 }
 
-// SetDNSName updates the DNS name for a given host. If the dnsName is empty,
-// this will no-op and will not unset the existing DNS name.
-func (h *Host) SetDNSName(ctx context.Context, dnsName string) error {
-	if h.Host == dnsName || dnsName == "" {
+// SetEC2Metadata updates the EC2 metadata for a given host. If the hostname on the input
+// params is empty, this will no-op and will not unset the existing DNS name.
+func (h *Host) SetEC2Metadata(ctx context.Context, params HostMetadataOptions) error {
```

Comment:
Now that this and the cloud host ready job are basically setting the same data, can we consolidate some of [this update](https://github.com/evergreen-ci/evergreen/blob/08b17dd4cf9697af6d461d0aec0ad19eb040f22f/model/host/host.go#L2139) with this function? Or reusing the existing cache function?